tests/static_multimap/static_multimap_test.cu

```diff
@@ -0,0 +1,298 @@
+                       (int32_t, dist_type::UNIQUE),
+                       (int64_t, dist_type::UNIQUE))
+{
+  using Key   = T;
+  using Value = T;
+
+  constexpr std::size_t num_keys{1'000'000};
+  cuco::static_multimap<Key, Value> map{2'000'000, -1, -1};
+
+  auto m_view = map.get_device_mutable_view();
+  auto view   = map.get_device_view();
+
+  std::vector<Key> h_keys(num_keys);
+  std::vector<cuco::pair_type<Key, Value>> h_pairs(num_keys);
+
+  generate_keys<Dist, Key>(h_keys.begin(), h_keys.end());
+
+  for (auto i = 0; i < num_keys; ++i) {
+    h_pairs[i].first  = h_keys[i] / 2;
+    h_pairs[i].second = h_keys[i];
+  }
+
+  thrust::device_vector<Key> d_keys(h_keys);
+  thrust::device_vector<cuco::pair_type<Key, Value>> d_pairs(h_pairs);
+
+  map.insert(d_pairs.begin(), d_pairs.end());
+
+  SECTION("Output of count and retrieve should be coherent.")
+  {
+    auto num = map.count(d_keys.begin(), d_keys.end());
+    thrust::device_vector<cuco::pair_type<Key, Value>> d_results(num);
+
+    REQUIRE(num == num_keys);
+
+    auto output_begin = d_results.data().get();
+    auto output_end   = map.retrieve(d_keys.begin(), d_keys.end(), output_begin);
+    auto size         = thrust::distance(output_begin, output_end);
+
+    REQUIRE(num == size);
+  }
+
+  SECTION("Output of count_outer and retrieve_outer should be coherent.")
+  {
+    auto num = map.count_outer(d_keys.begin(), d_keys.end());
+    thrust::device_vector<cuco::pair_type<Key, Value>> d_results(num);
+
+    REQUIRE(num == (num_keys + num_keys / 2));
+
+    auto output_begin = d_results.data().get();
+    auto output_end   = map.retrieve_outer(d_keys.begin(), d_keys.end(), output_begin);
+    auto size         = thrust::distance(output_begin, output_end);
+
+    REQUIRE(num == size);
+  }
+
+  SECTION("count_outer handles non-matches wile count doesn't.")
+  {
+    auto num_outer = map.count_outer(d_keys.begin(), d_keys.end());
+    auto num       = map.count(d_keys.begin(), d_keys.end());
+
+    REQUIRE(num_outer == (num + num_keys / 2));
+  }
+
+  SECTION("retrieve_outer handles non-matches wile retrieve doesn't.")
+  {
+    auto num_outer = map.count_outer(d_keys.begin(), d_keys.end());
+    thrust::device_vector<cuco::pair_type<Key, Value>> d_results(num_outer);
+
+    auto output_begin = d_results.data().get();
+    auto output_end   = map.retrieve(d_keys.begin(), d_keys.end(), output_begin);
+    auto size         = thrust::distance(output_begin, output_end);
+
+    output_end      = map.retrieve_outer(d_keys.begin(), d_keys.end(), output_begin);
+    auto size_outer = thrust::distance(output_begin, output_end);
+
+    REQUIRE(size_outer == (size + num_keys / 2));
+  }
+}
+
+TEMPLATE_TEST_CASE_SIG("Evaluation of pair functions",
+                       "",
+                       ((typename T, dist_type Dist), T, Dist),
+                       (int32_t, dist_type::UNIQUE),
+                       (int64_t, dist_type::UNIQUE))
+{
+  using Key   = T;
+  using Value = T;
```

Comment:
I see that all tests use key type == value type. Should we add 4B key/8B value and other combinations? (even if they fail, would be nice to test that and see if we get expected exception)